curriculum/challenges/english/25-front-end-development/quiz-css-variables/66ed9018f45ce3ece4053eb9.md

```diff
@@ -482,3 +482,465 @@ Which CSS gradient function allows you to create a gradient that radiates outwar
+
+### --question--
+
+#### --text--
+
+Which part of this `@property` declaration enforces type checking?
+
+```css
+@property --my-color {
+  syntax: "<color>";
+  inherits: false;
+  initial-value: red;
+}
+```
+
+#### --distractors--
+
+`initial-value`
+
+---
+
+`inherits`
+
+---
+
+`--my-color`
+
+#### --answer--
+
+`syntax`
+
+### --question--
+
+#### --text--
+
+What is the purpose of the `--gradient-angle` custom property in this example?
+
+```css
+.gradient {
+  background: linear-gradient(var(--gradient-angle), red, blue);
+}
+```
+
+#### --distractors--
+
+It defines the speed of the gradient transition.
+
+---
+
+It specifies the size of the gradient.
+
+---
+
+It sets the color blending mode.
+
+#### --answer--
+
+It controls the direction of the gradient.
+
+### --question--
+
+#### --text--
+
+Which of the following scenarios best demonstrates the power of custom properties for themes?
+
+#### --distractors--
+
+Using them only for font sizes.
+
+---
+
+Applying them only to `:hover` states.
+
+---
+
+Limiting them to one class.
+
+#### --answer--
+
+Defining light and dark theme values in `:root` and swapping them via class.
+
+### --question--
+
+#### --text--
+
+Why might developers prefer using `:root` for defining CSS variables?
+
+#### --distractors--
+
+Because `:root` improves page load speed.
+
+---
+
+Because `:root` disables specificity.
+
+---
+
+Because `:root` is required for JavaScript.
+
+#### --answer--
+
+Because `:root` allows the properties to be globally scoped.
+
+### --question--
+
+#### --text--
+
+What would the following CSS do if `--secondary-color` is undefined?
+
+```css
+h1 {
+  color: var(--secondary-color, orange);
+}
+```
+
+#### --distractors--
+
+It will apply `black`.
+
+---
+
+It will ignore the `color` property.
+
+---
+
+It will apply `--secondary-color`.
+
+#### --answer--
+
+It will apply `orange`.
+
+### --question--
+
+#### --text--
+
+What type of values does `<color>` represent in an `@property` syntax field?
+
+#### --distractors--
+
+Angles like `90deg`
+
+---
+
+Lengths like `10px`
+
+---
+
+Percentages like `50%`
+
+#### --answer--
+
+Color values like `#ff0000` or `red`
+
+### --question--
+
+#### --text--
+
+What is the main benefit of using custom properties in combination with `@property`?
+
+#### --distractors--
+
+They prevent needing fallback values.
+
+---
+
+They force static layout.
+
+---
+
+They reduce the need for classes.
+
+#### --answer--
+
+They enable smooth transitions and animations for custom properties.
+
+### --question--
+
+#### --text--
+
+What happens if you try to use a custom property inside a CSS rule that does not reference it properly?
+
+#### --distractors--
+
+The browser will ignore the entire rule.
+
+---
+
+The browser will throw an error.
+
+---
+
+The browser will apply the default value.
+
+#### --answer--
+
+The browser will ignore the custom property but apply the rest of the rule.
```

Comment:
I couldn't find where this specific question was covered. I don't know if campers will have enough information to answer this correctly.

So if that is the case, then this question needs to be updated.